bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/EditActionGroup.java

```
@@ -55,9 +57,22 @@ public EditActionGroup(Shell aShell) {
 
 	@Override
 	public void dispose() {
-		if (clipboard != null) {
-			clipboard.dispose();
-			clipboard = null;
+		Display display = Display.getCurrent();
+		if (display != null && !display.isDisposed()) {
+			if (clipboard != null && !clipboard.isDisposed()) {
+				clipboard.dispose();
+				clipboard = null;
+			}
+		} else {
+			// Non-UI thread or display disposed
+			if (clipboard != null && !clipboard.isDisposed()) {
+				try {
+					clipboard.dispose();
```

Can the clipboard really be disposed outside the UI thread?!?

---

Ideally not supposed to be which might result into SWTException or resource leaks if disposal fails silently or false positives. 
In my opinion - If we are not on the UI thread and can't guarantee access to it, it's safer to skip disposal or log a warning.

Do you mean i can skip this line `clipboard.dispose();` at line 70?